drivers/gnss/gnss_quectel_lx6.c

```diff
@@ -0,0 +1,732 @@
+		goto out_unlock;
+	}
+
+	ret = modem_chat_match_set_match(&data->pmtk_match, data->pmtk_match_buf);
+	if (ret < 0) {
+		goto out_unlock;
+	}
+
+	ret = modem_chat_run_script(&data->chat, &data->pmtk_script);
+	if (ret < 0) {
+		goto out_unlock;
+	}
+
+out_unlock:
+	(void)k_mutex_unlock(&data->mutex);
+
+	return ret;
+}
+
+static inline bool search_mode_enabled(const char *arg)
+{
+	return arg[0] == '1';
+}
+
+static int quectel_lx6_get_supported_systems(const struct device *dev, gnss_systems_t *systems)
+{
+	*systems = SUPPORTED_SYSTEMS;
+
+	return 0;
+}
+
+static DEVICE_API(gnss, gnss_api) = {
+	.set_fix_rate = quectel_lx6_set_fix_rate,
+	.get_fix_rate = NULL, /* not supported */
+	.set_navigation_mode = quectel_lx6_set_navigation_mode,
+	.get_navigation_mode = NULL, /* not supported */
+	.set_enabled_systems = quectel_lx6_set_enabled_systems,
+	.get_enabled_systems = NULL, /* not supported */
+	.get_supported_systems = quectel_lx6_get_supported_systems,
+};
+
+static int quectel_lx6_init_nmea0183_match(const struct device *dev)
+{
+	struct quectel_lx6_data *data = dev->data;
+
+	const struct gnss_nmea0183_match_config config = {
+		.gnss = dev,
+#if CONFIG_GNSS_SATELLITES
+		.satellites = data->satellites,
+		.satellites_size = ARRAY_SIZE(data->satellites),
+#endif
+	};
+
+	return gnss_nmea0183_match_init(&data->match_data, &config);
+}
+
+static void quectel_lx6_init_pipe(const struct device *dev)
+{
+	const struct quectel_lx6_config *config = dev->config;
+	struct quectel_lx6_data *data = dev->data;
+
+	if (config->i2c_bus) {
+#ifdef CONFIG_MODEM_BACKEND_QUECTEL_I2C
+		data->pipe = modem_backend_quectel_i2c_init(&data->backend.i2c,
+							    &config->backend.i2c_config);
+#endif
+	} else {
+#ifdef CONFIG_MODEM_BACKEND_UART
+		data->pipe = modem_backend_uart_init(&data->backend.uart,
+						     &config->backend.uart_config);
+#endif
+	}
+}
+
+static void quectel_lx6_init_pmtk_script(const struct device *dev)
+{
+	struct quectel_lx6_data *data = dev->data;
+
+	modem_chat_match_init(&data->pmtk_match);
+	modem_chat_match_set_separators(&data->pmtk_match, ",*");
+
+	modem_chat_script_chat_init(&data->pmtk_script_chat);
+	modem_chat_script_chat_set_response_matches(&data->pmtk_script_chat,
+						    &data->pmtk_match, 1);
+
+	modem_chat_script_init(&data->pmtk_script);
+	modem_chat_script_set_name(&data->pmtk_script, "pmtk");
+	modem_chat_script_set_script_chats(&data->pmtk_script, &data->pmtk_script_chat, 1);
+	modem_chat_script_set_abort_matches(&data->pmtk_script, NULL, 0);
+	modem_chat_script_set_timeout(&data->pmtk_script, 10);
+}
+
+static int quectel_lx6_init(const struct device *dev)
+{
+	const struct quectel_lx6_config *config = dev->config;
+	struct quectel_lx6_data *data = dev->data;
+	int ret;
+
+	k_mutex_init(&data->mutex);
+
+	if (config->vcc.port != NULL) {
+		ret = gpio_pin_configure_dt(&config->vcc, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("[%s] couldn't config VCC", dev->name);
+			return ret;
+		}
+	}
```

Comment:
This should be moved to the `quectel_lx6_turn_on` function, and in TURN_OFF the pin should probably be configured as an input to avoid back powering :) TURN_OFF means the power domain of the device will be turned off (if it has one)

Reply:
Done